app/uk/gov/hmrc/vatsubscription/models/post/MandationStatusPost.scala

```diff
@@ -16,13 +16,18 @@
 
 package uk.gov.hmrc.vatsubscription.models.post
 
-import play.api.libs.json.{Json, Reads}
+import play.api.libs.functional.syntax._
+import play.api.libs.json._
 import uk.gov.hmrc.vatsubscription.models.MandationStatus
+import uk.gov.hmrc.vatsubscription.models.MandationStatus.desReader
 
 case class MandationStatusPost(mandationStatus: MandationStatus,
                                transactorOrCapacitorEmail: Option[String])
 
 object MandationStatusPost {
 
-  implicit val reads: Reads[MandationStatusPost] = Json.reads[MandationStatusPost]
+  implicit val reads: Reads[MandationStatusPost] = (
+    (__ \ "mandationStatus").read[MandationStatus](desReader) and
```

Comment:
Does the reads here need to be dependent on the feature switch?

Reply:
This reads is just for the post from the opt-out service which will only ever send "3", so it doesn't matter which reads is used as both will work. I selected the newer one here purely so it's one less thing to change when the old one is removed.